modules/installing-gcp-user-defined-labels-and-tags.adoc

```diff
@@ -0,0 +1,126 @@
+// Module included in the following assemblies:
+// * installing/installing_gcp/installing-gcp-customizations.adoc
+
+:_content-type: CONCEPT
+[id="installing-gcp-user-defined-labels-and-tags_{context}"]
+= Configuring the user-defined labels and tags for GCP
+
+
+GCP provides tags that help to identify and organize the resources created for a specific {product-title} cluster, making them easier to manage. 
+
+You can define labels & tags on the GCP resources only during cluster creation that is available from {product-title} version {product-version}. User-defined labels and tags are not supported for {product-title} clusters upgraded to {product-title} version {product-version}.
+
+User-defined labels and {product-title} specific labels are applied only to resources created by {product-title} installer and its core operators such as `machine-api-provider-gcp`, `cluster-image-registry-operator`, `gcp-filestore-csi-driver-operator`, `gcp-pd-csi-driver-operator` and not on the resources created by any other operators or the kubernetes in-tree components that creates resources for example the ingress load balancers.
+
+User-defined tags are attached to resources created by {product-title} installer and its core operators such as `cluster-image-registry-operator` and not on the resources created by any other operators or the kubernetes in-tree components. In {product-title} version {product-version}, resources created by `machine-api-provider-gcp`, `gcp-filestore-csi-driver-operator`, `gcp-pd-csi-driver-operator` are not tagged.
+
+[NOTE]
+====
+Labels & tags once defined during the install-config creation phase cannot be modified or updated with new labels & tags after cluster creation.
+====
+
+Support for user-defined labels & tags for GCP resources is available in {product-title} version {product-version} as Technology Preview (TP).
+
+Service Account used for cluster creation must have the `TagUser` role assigned to it, required for tags defined at both organization and project level.
```

Comment:
`TagViewer` role is also required. Will wait for @jianli-wei confirmation.

```suggestion
Service Account used for cluster creation must have the `TagViewer` and `TagUser` roles assigned to it, required for tags defined at both organization and project level.
```

Reply:
Ack.

Reply:
@subhtk Yes, I was trying the installations using 4.14 nightly latest yesterday, unfortunately failed (see [comment](https://issues.redhat.com/browse/CORS-2536?focusedId=22849021&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-22849021)), I guess more debugging needed. 

Maybe you could move on now, and we could use another PR to correct the required roles if necessary. Thanks!

Reply:
@subhtk I did some tests as below, and I think we could remove `Tag Viewer` role here. 

You can see `Tag User` role has all the permissions of `Tag Viewer` role, the only exception is `resourcemanager.tagHolds.list` permission, which has been included in the pre-defined role `Security Admin` (or IAM Security Admin), which is in turn documented already (see [OCP doc](https://docs.openshift.com/container-platform/4.13/installing/installing_gcp/installing-gcp-account.html#installation-gcp-permissions_installing-gcp-account) section "_Required roles for the installation program_").

```
$ gcloud iam roles describe roles/resourcemanager.tagUser > /tmp/resourcemanager.tagUser
$ gcloud iam roles describe roles/resourcemanager.tagViewer > /tmp/resourcemanager.tagViewer
$ diff /tmp/resourcemanager.tagUser /tmp/resourcemanager.tagViewer | grep -E "^> "
> description: Access to list Tags and their associations with resources
> - resourcemanager.tagHolds.list
> name: roles/resourcemanager.tagViewer
> title: Tag Viewer
$ gcloud iam roles describe roles/iam.securityAdmin | grep resourcemanager.tagHolds.list
- resourcemanager.tagHolds.list
$ 
```

I tried with a QE SA, after removing `Tag Viewer` role from the SA, to list tag keys/values under Org level or QE projects level still works well. @bharath-b-rh Please advise, thanks!

Reply:
@jianli-wei Thank you for confirming! Yeah `TagUser` supersedes `TagViewer`, since during tests we faced some issues thought it was also required.
@subhtk Please disregard my comment to add `TagViewer` role.

Reply:
Okay.

Reply:
@bharath-b-rh Done with the changes, have a final review?